model_analyzer/model_manager.py

```diff
@@ -0,0 +1,269 @@
+from model_analyzer.config.run.run_search import RunSearch
+from model_analyzer.config.run.run_config_generator \
+    import RunConfigGenerator
+from model_analyzer.model_analyzer_exceptions \
+    import TritonModelAnalyzerException
+
+import logging
+import os
+import heapq
+import shutil
+
+
+class ModelManager:
+    """
+    This class handles the search for, creation of, and execution of run configs.
+    It also records the best results for each model.
+    """
+    def __init__(self, config, client, server, metrics_manager,
+                 result_manager):
+        """
+        Parameters
+        ----------
+        config: AnalyzerConfig
+            The config for the model analyzer
+        """
+
+        self._config = config
+        self._client = client
+        self._server = server
+        self._metrics_manager = metrics_manager
+        self._result_manager = result_manager
+        self._run_search = RunSearch(config=config)
+        self._run_config_generator = RunConfigGenerator(config=config,
+                                                        client=self._client)
+
+        # Generate the output model repository path folder.
+        self._output_model_repo_path = config.output_model_repository_path
+        try:
+            os.mkdir(self._output_model_repo_path)
+        except OSError:
+            if not config.override_output_model_repository:
+                raise TritonModelAnalyzerException(
+                    f'Path "{self._output_model_repo_path}" already exists. '
+                    'Please set or modify "--output-model-repository-path" flag or remove this directory.'
+                    ' You can also allow overriding of the output directory using'
+                    ' the "--override-output-model-repository" flag.')
+            else:
+                shutil.rmtree(self._output_model_repo_path)
+                logging.warn(
+                    f'Overriding the output model repo path "{self._output_model_repo_path}"...'
+                )
+                os.mkdir(self._output_model_repo_path)
+
+        # Model results
+        self._model_results = []
+
+    def run_model(self, model):
+        """
+        Generates configs, runs inferences, gets
+        measurements for a single model
+
+        Parameters
+        ----------
+        model : ConfigModel
+            The model being run
+        """
+
+        # Clear any configs from previous model run
+        self._run_config_generator.clear_configs()
+
+        if self._config.run_config_search_disable:
+            self._run_model_no_search(model)
+        else:
+            self._run_model_with_search(model)
+
+        # Sort the results for this model
+        self._result_manager.sort_results()
+
+        # Save the best result
+        [top_result] = self._result_manager.top_n_results(n=1)
+        heapq.heappush(self._model_results, (top_result, model.model_name()))
+
+    def top_n_models(self, n):
```

Comment:
Is this `top_n_models` or `top_n_configs` acrosss all the models.

Reply:
I think the Bermuda team requested the top n configs across all the models.

Reply:
From a list of the best configs from each model, this chooses the top n configs.